xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,38 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/// Stopper subclass which checks for both stemmed and unstemmed stopwords
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+    std::unordered_set<std::string> stop_words;
+    Xapian::Stem stemmer;
+
+  public:
+    /** Constructor
+     *
+     *  @param stemmer	The Xapian::Stem object to set.
+     */
+    explicit StemStopper(const Xapian::Stem &stemmer);
+
+    std::string get_description() const;
+
+    bool operator()(const std::string & term) const {
+	return stop_words.find(term) != stop_words.end();
+    }
+
+    /// Add a single stop word and its stemmed equivalent
+    void add(const std::string &term);
+};
+
```

Comment:
> We can always check if it's a stopword before stemming

The plan is to use a `Stopper` here to allow filtering out already-stemmed terms from documents we're about to cluster.  The stemming happened at index time, so we can't really check before stemming in this case.

> I don't see any changes where it's used in this pull request, so curious is what's the use-case.

Hmm, there really ought to be a feature test added for `StemStopper` to ensure it works as intended, and continues to do so.